include/sycl/algorithm/inner_product.hpp

```diff
@@ -34,7 +34,9 @@
 #include <iostream>
 
 #include <sycl/algorithm/algorithm_composite_patterns.hpp>
+#include <sycl/algorithm/reduce.hpp>
```

Comment:
Why should this be an issue, if it is internal?
I am more puzzled by the
```
#include <iostream>
```
why is there a need for I/O in something not related to I/O?